docs/dev/rfc/odfe-cli-rest-api-as-commands.md

```diff
@@ -0,0 +1,298 @@
+# RFC: Support REST api as command
+
+## Objective
+
+This document describes how to support [rest api](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html) in a generic way.
+As of now, elasticsearch support api in different categories like index api, cat api, cluster api, document api, etc to
+configure and access their features. It will take tremendous amount of time and effort to provide every api as native commands.
+Hence, we will be categorizing those REST API based on GET/PUT/POST/DELETE  and allow users to perform their request 
+in a generic way. This will support any future API without any additional support. It also benefits plugin owners while on-boarding their plugin. 
```

Comment:
Benefitting plugin owners isn't exactly clear to me. How will this change anything?

Reply:
So right now, plug in owners has to build the gateway layer by using framework methods like build request/ call request. Now, they can call controller methods directly by calling get/put/post/delete method . Since their API can be represented as controller methods.

Reply:
Can you add that explanation here?

Reply:
on second thought, removed this point. I will update our developer guideline to include this benefit once i merge this feature, since we need to provide a new direction on how to use this in framework. I am planning to keep those discussion away from this feature at this point.